### PR TITLE
Abstract queryId generation

### PR DESCRIPTION
**Pull request by ykmr1224**
### Description
- Abstract queryId generation
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spark/src/main/java/org/opensearch/sql/spark/asyncquery/model/AsyncQueryJobMetadata.java [hunk 1]
```diff
@@ -59,6 +59,6 @@ public static AsyncQueryJobMetadata copy(
 
   @Override
   public String getId() {
-    return queryId.docId();
+    return "qid" + queryId;
```
Comment (seankao-az) [inline spark/src/main/java/org/opensearch/sql/spark/asyncquery/model/AsyncQueryJobMetadata.java:62]: why add "qid"? is this for OpenSearch doc only?

File: spark/src/main/java/org/opensearch/sql/spark/dispatcher/DatasourceEmbeddedQueryIdProvider.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import org.opensearch.sql.spark.asyncquery.model.AsyncQueryId;
+import org.opensearch.sql.spark.dispatcher.model.DispatchQueryRequest;
+
+/** Generates QueryId by embedding Datasource name and random UUID */
+public class DatasourceEmbeddedQueryIdProvider implements QueryIdProvider {
+
+  @Override
+  public String getQueryId(DispatchQueryRequest dispatchQueryRequest) {
+    return AsyncQueryId.newAsyncQueryId(dispatchQueryRequest.getDatasource()).getId();
```
Comment (seankao-az) [inline spark/src/main/java/org/opensearch/sql/spark/dispatcher/DatasourceEmbeddedQueryIdProvider.java:16]: `AsyncQueryId` is only used here? Could you consider rename it to `DatasourceEmbeddedQueryId`? Or is it also used elsewhere?

File: spark/src/main/java/org/opensearch/sql/spark/dispatcher/QueryIdProvider.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.dispatcher;
+
+import org.opensearch.sql.spark.dispatcher.model.DispatchQueryRequest;
+
+/** Interface for extension point to specify queryId. Called when new query is executed. */
+public interface QueryIdProvider {
+  String getQueryId(DispatchQueryRequest dispatchQueryRequest);
```
Comment (seankao-az) [inline spark/src/main/java/org/opensearch/sql/spark/dispatcher/QueryIdProvider.java:12]: is this parameter sufficient for generating query id? or are we expecting implementation to use thread context to pass them

File: spark/src/main/java/org/opensearch/sql/spark/asyncquery/model/AsyncQueryJobMetadata.java (hunk 1 above)
Comment (ykmr1224) [reply spark/src/main/java/org/opensearch/sql/spark/asyncquery/model/AsyncQueryJobMetadata.java:62]: It is to keep the same logic.
It was originally implemented in AsyncQueryId class.

File: spark/src/main/java/org/opensearch/sql/spark/dispatcher/DatasourceEmbeddedQueryIdProvider.java (hunk 2 above)
Comment (ykmr1224) [reply spark/src/main/java/org/opensearch/sql/spark/dispatcher/DatasourceEmbeddedQueryIdProvider.java:16]: Good point. Will fix.

File: spark/src/main/java/org/opensearch/sql/spark/dispatcher/QueryIdProvider.java (hunk 3 above)
Comment (ykmr1224) [reply spark/src/main/java/org/opensearch/sql/spark/dispatcher/QueryIdProvider.java:12]: It should be sufficient.

**Review by seankao-az — APPROVED** [review]

**Review by rupal-bq — APPROVED** [review]

**Pull request merged**